ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/channels/RangedSeekableByteChannel.java

```diff
@@ -0,0 +1,196 @@
+            final BulkObject previous = sortedList.get(i-1);
+
+            final long previousOffset = realOffsets.get(previous);
+            final long previousSize = blobLengths.get(previous);
+
+            realOffsets.put(sortedList.get(i), previousOffset + previousSize);
+        }
+
+        return ImmutableMap.copyOf(realOffsets);
+    }
+
+    private static ImmutableMap<BulkObject, Long> computesBlobSize(final ImmutableMultimap<BulkObject, Range> ranges) {
+        if (Guard.isMultiMapNullOrEmpty(ranges)) {
+            return ImmutableMap.of();
+        }
+
+        final ImmutableMap.Builder<BulkObject, Long> builder = ImmutableMap.builder();
+
+        for (final BulkObject blob : ranges.keySet()) {
+            builder.put(blob, sizeOfBulkRange(ranges.get(blob)));
+        }
+
+        return builder.build();
+    }
+
+    private static long getSize(final long channelSize, final ImmutableMultimap<BulkObject, Range> ranges) {
+        if (Guard.isMultiMapNullOrEmpty(ranges)) {
+            return channelSize;
+        } else {
+            return sizeOfBulkRange(ranges.values());
+        }
+    }
+
+    private static long sizeOfBulkRange(final ImmutableCollection<Range> ranges) {
+        long rangeSize = 0;
+        for (final Range range : ranges) {
+            rangeSize += range.getLength();
+        }
+        return rangeSize;
+    }
+
+    @Override
+    public int read(final ByteBuffer dst) throws IOException {
+        checkClosed();
+        final int bytesRead = byteChannel.read(dst);
+        this.position += bytesRead;
+        return bytesRead;
+    }
+
+    @Override
+    public int write(final ByteBuffer src) throws IOException {
+        checkClosed();
+        final int bytesWritten = byteChannel.write(src);
+        this.position += bytesWritten;
+        return bytesWritten;
+    }
+
+    @Override
+    public long position() throws IOException {
+        checkClosed();
+        return this.position;
+    }
+
+    @Override
+    public SeekableByteChannel position(final long newPosition) throws IOException {
+        checkClosed();
+        if (!checkRange(newPosition)) {
+            throw new IllegalStateException("The requested position is outside the acceptable ranges for this stream");
+        }
+        this.position = newPosition;
+
+        final BulkObject blob = getBlobFromPosition(ranges, newPosition);
+        if (blob == null) {
+            byteChannel.position(newPosition);
+        } else {
+            final long offsetInBlob = newPosition - blob.getOffset();
+            final long blobOffset = startingOffsetForBlob.get(blob);
+
+            byteChannel.position(blobOffset + offsetInBlob);
+        }
+
+        return this;
+    }
+
+    private boolean checkRange(final long position) {
+        if (Guard.isMultiMapNullOrEmpty(ranges)) return true;  // this means we are reading from the stream and we do not any range checks
```

Comment:
Comment is grammatically incorrect.  Suggestion:
"... we do not need any range checks"